Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5703,6 +5703,8 @@ void WebPageProxy::didCommitLoadForFrame(FrameIdentifier frameID, FrameInfoData&
 #endif
 #if ENABLE(NETWORK_CONNECTION_INTEGRITY)
         updateAllowedLookalikeCharacterStringsIfNeeded();
+        if (navigation && navigation->websitePolicies())
+            m_networkConnectionIntegrityPolicies = navigation->websitePolicies()->networkConnectionIntegrityPolicy();
```

Comment:
Nit - can we also reset `m_networkConnectionIntegrityPolicies` in `WebPageProxy::resetState`?